src/app/shared/base-value-accessor/base-value-accessor.directive.ts

```diff
@@ -0,0 +1,43 @@
+import { AfterViewInit, Directive, Input, ViewChild } from '@angular/core';
+import { ControlValueAccessor, DefaultValueAccessor } from '@angular/forms';
+import { ReplaySubject } from 'rxjs';
+
+@Directive({
+  selector: 'base-value-accessor',
+})
+export class BaseValueAccessorDirective
```

Comment:
I don't understand why it's directive. If you want to put some common logic into a base class, then it should be a BaseValueAccessorClass.